context/src/main/java/io/grpc/Deadline.java

```diff
@@ -52,24 +51,25 @@
    * @return A new deadline.
    */
   public static Deadline after(long duration, TimeUnit units) {
-    return after(duration, units, SYSTEM_TICKER);
+    checkNotNull(units, "units");
+    return new Deadline(SystemTicker.INTANCE, units.toNanos(duration), true);
   }
 
-  // For testing
-  static Deadline after(long duration, TimeUnit units, Ticker ticker) {
+  //For testing
+  public static Deadline forTest(long duration, TimeUnit units, DeadlineTicker ticker) {
```

Comment:
Isn't this constructor hard to use in code without an easily accessible instance of SystemTicker? Although it sounds like you've created your own custom system ticker that is both a DeadlineTicker and cgc.base.Ticker.